src/app/core/services/connector-limits.service.ts

```diff
@@ -0,0 +1,49 @@
+import {Injectable} from '@angular/core';
+import {Observable, of} from 'rxjs';
+import {map} from 'rxjs/operators';
+import {ConnectorLimits} from '@sovity.de/edc-client';
+import {ActiveFeatureSet} from '../config/active-feature-set';
+import {EdcApiService} from './api/edc-api.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConnectorLimitsService {
+  constructor(
+    private edcApiService: EdcApiService,
+    private activeFeatureSet: ActiveFeatureSet,
+  ) {}
+
+  fetchLimits(): Observable<ConnectorLimits> {
+    return this.edcApiService.getEnterpriseEditionConnectorLimits();
+  }
+
+  limitExceeded(): Observable<boolean> {
+    return this.fetchLimits().pipe(
+      map((limits) => {
+        if (
+          limits.maxActiveConsumingContractAgreements === null ||
+          limits.maxActiveConsumingContractAgreements === undefined
+        ) {
+          return false;
+        }
+        return (
+          limits.numActiveConsumingContractAgreements >=
+          limits.maxActiveConsumingContractAgreements
+        );
+      }),
+    );
+  }
+
+  canNegotiate(): Observable<boolean> {
+    if (!this.activeFeatureSet.hasConnectorLimits()) {
+      return of(true);
+    }
```

Comment:
i believe you can move this to the limits exceeded method and skip the inversion method "canNegotiate". It's always a question of design and convenience in which negation you have your methods. Consistency can be something that helps you decide, but I'd argue we don't need two methods, just to have one be a renamed method with the inversion. You can use the "isConsumingAgreementLimitExceeded" directly, you can ofc assign it in however it is useful to you, e.g. "disableNegotiate" or whatever in the component, right?